### PR TITLE
Bump `pytest-asyncio` from 0.15.1 to 0.18.2

### PR DESCRIPTION
**Pull request by Kludex**
Replaces #1408

~This PR adds the `asyncio_mode=auto` option as well. Which means that `@pytest.mark.asyncio` markers can be removed.~

~Previous approach raised errors, which I don't have the time to solve. In any case, this PR just bumps `pytest-asyncio`, and adds the configuration that is needed to match the behavior we had on 0.15.1.~

Both approaches fail with the same error. :disappointed:

### CODE REVIEW

**Comment by euri10** [comment]
I'm game looking at this

**Comment by euri10** [comment]
ok I spent 2h and I'm nowhere close to understand what the f** happens on windows, those warnings about unclosed resources are insanely hard to debug

**Comment by euri10** [comment]
the thing I really dont get is why it fails with anyio ??
`E               KeyError: 'anyio._backends._asyncio'`

**Comment by euri10** [comment]
I think the root issue may be that on windows the closing of the ssl context is too slow but I'm not yet entirely sure

File: requirements.txt [hunk 1]
```diff
@@ -22,7 +22,7 @@ cryptography==3.4.8
 coverage==5.5
 coverage-conditional-plugin==0.4.0
 httpx==1.0.0b0
-pytest-asyncio==0.15.1
+pytest-asyncio==0.18.2
```
Comment (Kludex) [inline requirements.txt]:
```suggestion
pytest-asyncio==0.18.3
```

**Comment by agronholm** [comment]
I recall there was a bug regarding unclosed sockets in asyncio that was fixed in later versions of CPython. I do wonder why you're only hitting it now though.

**Comment by euri10** [comment]
> I recall there was a bug regarding unclosed sockets in asyncio that was fixed in later versions of CPython. I do wonder why you're only hitting it now though.

and only on windows, and it differs between 3.8 and 3.9
will try to find that report @agronholm but if you find it before dont hesitate :)

**Comment by Kludex** [comment]
Hmmm... The 3.8 versions are different between those two OS: windows is using 3.8.10, ubuntu is using 3.8.12.

**Comment by agronholm** [comment]
I've just ignored those in the AnyIO test suite: https://github.com/agronholm/anyio/blob/master/pyproject.toml#L41

**Comment by Kludex** [comment]
There's no 3.8.12 for Windows yet...

File: .github/workflows/test-suite.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
     runs-on: "${{ matrix.os }}"
     strategy:
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        python-version: ["3.7", "3.8.12", "3.9", "3.10"]
```
Comment (Kludex) [inline .github/workflows/test-suite.yml]:
```suggestion
        python-version: ["3.7", "3.8", "3.9", "3.10"]
```

**Comment by euri10** [comment]
> I've just ignored those in the AnyIO test suite: https://github.com/agronholm/anyio/blob/master/pyproject.toml#L41

probably the smart move, I'd like though to do that only on windows since we have zero issues on macos and linux

**Comment by Kludex** [comment]
Let's drop `pytest-asyncio` and use only `anyio`, as we already have this dependency via `httpx`.

- See https://github.com/encode/uvicorn/pull/1155.

**Pull request closed without merging**